src/lib/utils/arweaveUtils.ts

```diff
@@ -1,47 +1,54 @@
 import * as B64js from "base64-js";
 
-export type Base64UrlString = string;
-
-export function concatBuffers(
-  buffers: Uint8Array[] | ArrayBuffer[],
-): Uint8Array {
-  let total_length = 0;
-
-  for (let i = 0; i < buffers.length; i++) {
-    total_length += buffers[i].byteLength;
-  }
-
-  let temp = new Uint8Array(total_length);
-  let offset = 0;
+// BINARY DATA TYPES:
 
-  temp.set(new Uint8Array(buffers[0]), offset);
-  offset += buffers[0].byteLength;
-
-  for (let i = 1; i < buffers.length; i++) {
-    temp.set(new Uint8Array(buffers[i]), offset);
-    offset += buffers[i].byteLength;
-  }
+export type Base64UrlString = string;
 
-  return temp;
+export type TypedArray =
+  | Int8Array
+  | Uint8Array
+  | Uint8ClampedArray
+  | Int16Array
+  | Uint16Array
+  | Int32Array
+  | Uint32Array
+  // | Float16Array
+  | Float32Array
+  | Float64Array
+  | BigInt64Array
+  | BigUint64Array;
+
+export type BinaryDataType = ArrayBuffer | TypedArray | DataView | Buffer;
+
+export function binaryDataTypeToString(buffer: BinaryDataType): string {
+  return new TextDecoder("utf-8", { fatal: true }).decode(buffer);
 }
 
-export function b64UrlToString(b64UrlString: string): string {
-  let buffer = b64UrlToBuffer(b64UrlString);
-
-  return bufferToString(buffer);
+export function binaryDataTypeOrStringToString(
+  source: string | BinaryDataType,
+) {
+  return typeof source === "string" ? source : binaryDataTypeToString(source);
 }
 
-export function bufferToString(buffer: Uint8Array | ArrayBuffer): string {
-  return new TextDecoder("utf-8", { fatal: true }).decode(buffer);
+export function binaryDataTypeOrStringToBinaryDataType(
+  source: string | BinaryDataType,
+) {
+  return typeof source === "string" ? stringToUint8Array(source) : source;
 }
 
-export function stringToBuffer(string: string): Uint8Array {
+export function stringToUint8Array(string: string): Uint8Array {
```

Comment:
@7i7o I've updated the names to be more explicit as the main confusion when looking at the old code came from operations with the binary data types.

Also, this seems to be an issue in ArConnect too, as if you look at the README, type definitions (repo) and documentation, there are discrepancies that, while probably work just fine in practice, might be misleading to developers and cause issues where TS reports everything is fine, but they get runtime errors trying to work with the binary data type assuming an incorrect type.

Reply:
Yes, we talked about this. I agree that we needed better naming. Great catch about the Arconnect ones as well! We can plan a PR for that later on as well